arangod/Pregel/REST/RestPregelHandler.cpp

```diff
@@ -74,9 +76,42 @@ RestStatus RestPregelHandler::execute() {
         return RestStatus::DONE;
       }
       if (msg.get().receiver.id == actor::ActorID{0}) {
+        auto spawnMessage =
+            inspection::deserializeWithErrorT<message::SpawnMessages>(
+                velocypack::SharedSlice({}, msg.get().payload.slice()));
+        if (!spawnMessage.ok()) {
+          generateError(
+              rest::ResponseCode::BAD, TRI_ERROR_BAD_PARAMETER,
+              fmt::format(
+                  "Received actor spawn message {} cannot be deserialized",
+                  body.toJson()));
+          return RestStatus::DONE;
+        }
+        if (!std::holds_alternative<message::SpawnWorker>(spawnMessage.get())) {
+          generateError(rest::ResponseCode::BAD, TRI_ERROR_BAD_PARAMETER,
+                        fmt::format("Received actor spawn message {} does not "
+                                    "include expected spawn worker message",
+                                    body.toJson()));
+          return RestStatus::DONE;
+        }
+
+        // ActorID "0" being used here to initially spawn a new actor.
+        auto resultActorID = _pregel._actorRuntime->spawn<ResultActor>(
+            _vocbase.name(), std::make_unique<ResultState>(),
+            message::ResultMessages{message::ResultStart{}});
+        auto resultActorPID =
+            actor::ActorPID{.server = ServerState::instance()->getId(),
+                            .database = _vocbase.name(),
+                            .id = resultActorID};
```

Comment:
For your intererest: I added ticket https://arangodb.atlassian.net/browse/GORDO-1576 to clean-up the actor, including that the spawn function should return an ActorPID instead of an ActorID, such that we don't have to create the PID here by ourselves.